option.go

```diff
@@ -157,6 +159,8 @@ func (s *Server) setNetworkBootOpts(ctx context.Context, m *dhcpv4.DHCPv4, n *da
 }
 
 // bootfileAndNextServer returns the bootfile (string) and next server (net.IP).
+// input arguments `tftp`, `ipxe` and `iscript` use non string types so as to attempt to be more clear about the expectation around what is wanted for these values.
+// It also helps us avoid having to validate a string in multiple ways.
```

Comment:
These are helpful, but I also curious about the formatting being done to generate things like `bootfile` and `bin`.
I should have been more clear.

Reply:
Per our verbal conversation, this is around how we're wiring up "system level" observability by encoding interesting information in our responses. I think you'll address that later, so perhaps we can just merge this PR as-is.